client/config.go

```diff
@@ -0,0 +1,44 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+// SetConfig requests a snap to set the provided config.
+func (client *Client) SetConfig(snapName string, config map[string]interface{}) (changeID string, err error) {
```

Comment:
Reading this PR made me wonder: since we're shortening the word anyway, can we try to build the feature everywhere around the "conf" term instead of "config"? It's nice, shorter, and reminds of ".conf".

Also, can we please call the parameter above "patch" rather than "conf" or "config", to make it clear that this is not a full config?